layer/logged_data/system_metrics.py

```diff
@@ -84,23 +105,86 @@ def get_memory_metrics(self) -> Dict[str, Dict[str, float]]:
         }
 
 
+class CGroupsMetricsCollectorV1(CGroupsMetricsCollector):
+    """
+    Use cgroups-v1 to collect system metrics
+    """
+
+    def get_cpu_usage(self) -> int:
+        # Time in nanoseconds
+        return int(self._read_cgroup_metric("cpu/cpuacct.usage_user"))
+
+    def get_cpu_available(self) -> float:
+        # Times in microseconds
+        cpu_quota = int(self._read_cgroup_metric("cpu/cpu.cfs_quota_us"))
+        cpu_period = int(self._read_cgroup_metric("cpu/cpu.cfs_period_us"))
+        return float(cpu_quota / cpu_period)
+
+    def get_mem_used(self) -> int:
+        return int(self._read_cgroup_metric("memory/memory.usage_in_bytes"))
+
+    def get_mem_available(self) -> int:
+        return int(self._read_cgroup_metric("memory/memory.limit_in_bytes"))
+
+
+class CGroupsMetricsCollectorV2(CGroupsMetricsCollector):
+    """
+    Use cgroups-v2 to collect system metrics
+    """
+
+    def get_cpu_usage(self) -> int:
+        # Time expected in nanoseconds
+        cpu_usage = str(self._read_cgroup_metric("user.slice/cpu.stat"))
+        regex = r"user_usec (\d*)\n"
+        extracted_cpu_usage = re.search(regex, cpu_usage)
+        if extracted_cpu_usage is None:
+            return 0
+        else:
+            # Time in microseconds, converting to nanoseconds
+            return int(extracted_cpu_usage.group(1)) * 1000
+
+    def get_cpu_available(self) -> float:
+        max_cpu = str(self._read_cgroup_metric("user.slice/cpu.max"))
+        cpu_quota = max_cpu.split()[0]
+        cpu_period = max_cpu.split()[1]
+        if str(cpu_quota) == "max":
+            return 1
+        else:
+            return float(int(cpu_quota) / int(cpu_period))
+
+    def get_mem_used(self) -> int:
+        return int(self._read_cgroup_metric("user.slice/memory.current"))
+
+    def get_mem_available(self) -> int:
+        memory_high = self._read_cgroup_metric("user.slice/memory.high")
+        if memory_high == "max":
+            return int(psutil.virtual_memory().total)
+        else:
+            return int(memory_high)
+
+
+class CGroupsVersion(Enum):
+    V1 = "v1"
+    V2 = "v2"
+    UNKNOWN = "unknown"
+
+
 class DockerMetricsCollector(MetricsCollector):
```

Comment:
Can we rename this to CGroupsMetricsCollector? I guess in the case of Ubuntu, we will be reading CGroups, but there is not necessarily docker involved there.